math/src/polynom/multi_linear.rs

```diff
@@ -0,0 +1,240 @@
+use alloc::vec::Vec;
+use core::ops::Index;
+
+use utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
+
+use crate::FieldElement;
+
+// MULTI-LINEAR POLYNOMIAL
+// ================================================================================================
+
+/// Represents a multi-linear polynomial.
+///
+/// The representation stores the evaluations of the polynomial over the boolean hyper-cube
+/// {0 , 1}^ν.
+#[derive(Clone, Debug)]
+pub struct MultiLinearPoly<E: FieldElement> {
+    num_variables: usize,
+    evaluations: Vec<E>,
+}
+
+impl<E: FieldElement> MultiLinearPoly<E> {
+    /// Constructs a [`MultiLinearPoly`] from its evaluations over the boolean hyper-cube {0 , 1}^ν.
+    pub fn from_evaluations(evaluations: Vec<E>) -> Result<Self, MultiLinearPolyError> {
+        if !evaluations.len().is_power_of_two() {
+            return Err(MultiLinearPolyError::EvaluationsNotPowerOfTwo);
+        }
+        Ok(Self {
+            num_variables: (evaluations.len().ilog2()) as usize,
+            evaluations,
+        })
+    }
```

Comment:
I think it is OK to panic here is the number of evaluations is not a power of two. This way, we also won't need the `MultiLinearPolyError` enum.

Reply:
Changed